http/client/src/main/java/org/wildfly/security/http/client/utils/DigestHttpMechanismUtil.java

```diff
@@ -0,0 +1,146 @@
+package org.wildfly.security.http.client.utils;
+
+import org.wildfly.security.digest.WildFlyElytronDigestProvider;
+import org.wildfly.security.http.client.exception.ElytronHttpClientException;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpRequest;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class DigestHttpMechanismUtil {
+
+    public HttpRequest createDigestRequest(String uri, String userName, String password, Map<String,String> authParams) throws Exception {
+        String realm = authParams.getOrDefault("realm", null);
+        String domain = authParams.getOrDefault("domain", null);
+        String nonce = authParams.getOrDefault("nonce", null);
+        String opaque = authParams.getOrDefault("opaque", null);
+        String algorithm = authParams.getOrDefault("algorithm", "MD5");
+        String qop = authParams.getOrDefault("qop", null);
+        String uriPath = getUriPath(uri);
+        String cnonce = generateCNonce();
+        String nc = String.format("%08x", Integer.parseInt(authParams.getOrDefault("nc", String.valueOf(0))));
+
+        String resp;
+        if (qop == null) {
+            resp = computeDigestWithoutQop(uriPath, nonce, userName, password, algorithm, realm, "GET");
+        } else {
+            resp = computeDigestWithQop(uriPath, nonce, cnonce, nc, userName, password, algorithm, realm, qop, "GET");
+        }
+
+        HttpRequest request = HttpRequest
+                .newBuilder()
+                .uri(new URI(uri))
+                .header("Authorization", "Digest " +
+                        "username=\"" + userName + "\", " +
+                        "realm=\"" + realm + "\"," +
+                        "nonce=\"" + nonce + "\", " +
+                        "uri=\"" + uriPath + "\", " +
+                        "qop=\"" + qop + "\", " +
+                        "nc=\"" + nc + "\"," +
+                        "cnonce=\"" + cnonce + "\", " +
+                        "response=\"" + resp + "\", " +
+                        "opaque=\"" + opaque + "\", " +
+                        "algorithm=" + algorithm)
+                .build();
+        int updateNonceCount = Integer.parseInt(authParams.getOrDefault("nc", "0")) + 1;
+        authParams.put("nc", String.valueOf(updateNonceCount));
+        return request;
+    }
+
+    public void updateAuthParams(String authHeader, Map<String,String> authParams) {
+        Pattern realmPattern = Pattern.compile("realm=\"(.*?)\"");
+        Pattern domainPattern = Pattern.compile("domain=\"(.*?)\"");
+        Pattern noncePattern = Pattern.compile("nonce=\"(.*?)\"");
+        Pattern opaquePattern = Pattern.compile("opaque=\"(.*?)\"");
+        Pattern algorithmPattern = Pattern.compile("algorithm=(.+?)(?:,|$)");
+        Pattern qopPattern = Pattern.compile("qop=\\s*\"?([^\"]*)\"?");
```

Comment:
@keshav-725 We should not use the regexes here. It is error prone and it does not check the full header that could contain other fields/information in it. Take a look at the DigestSaslClient to see how the parsing is being done there

Reply:
@keshav-725 You can use the same utility method for parsing that is being used by sasl client mechanism by adding a dependency on `wildfly-elytron-mechanism-digest` module

Reply:
This is the method: https://github.com/wildfly-security/wildfly-elytron/blob/1.x/mechanism/digest/src/main/java/org/wildfly/security/mechanism/digest/DigestUtil.java#L59